packages/blockchain-utils-linking/src/ethereum.ts

```diff
@@ -61,10 +63,40 @@ export class EthereumAuthProvider implements AuthProvider {
     return createLink(did, accountId, this.provider, this.opts)
   }
 
-  async requestCapability(siweMessage: SiweMessage): Promise<Cacao> {
+  async requestCapability(
+    sessionDID: string,
+    streams: Array<StreamID | string>,
+    opts: CapabilityOpts
+  ): Promise<Cacao> {
     console.warn(
       'WARN: requestCapability os an experimental API, that is subject to change any time.'
     )
+
+    const domain = typeof window !== 'undefined' ? window.location.hostname : opts.domain
+    if (!domain) throw new Error("Missing parameter 'domain'")
+
+    // NOTE: To allow proper customization of the expiry date, we need a solid library to represent
+    // time durations that includes edge cases. We should not try dealing with timestamps ourselves.
+    const now = new Date()
+    const oneWeekLater = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000)
+
+    const siweMessage = new SiweMessage({
+      domain: domain,
+      address: this.address,
+      statement: opts.statement
+        ? opts.statement
+        : 'Give this application access to some of your data on Ceramic',
+      uri: sessionDID,
+      version: opts.version ? opts.version : '1',
+      nonce: opts.nonce ? opts.nonce : randomString(10),
```

Comment:
```suggestion
      statement: opts.statement || 'Give this application access to some of your data on Ceramic',
      uri: sessionDID,
      version: opts.version || '1',
      nonce: opts.nonce || randomString(10),
```

Reply:
reminded me of this - https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx
off the top of my head thought it also applied to `||`
perhaps a stronger approach is to use `??`

Reply:
Thoughts on this @ukstv ?

Reply:
IMO, it makes sense to leave `||` assuming we can rely on TS type system.

Reply:
going to remove the ternary but going to use `??`